src/join/joining.cu

```diff
@@ -219,15 +165,117 @@ gdf_error sort_join<JoinType::LEFT_JOIN>(gdf_column *leftcol, gdf_column *rightc
                                          gdf_column *l_result, gdf_column *r_result,
                                          gdf_context *ctxt);
 
+/* --------------------------------------------------------------------------*/
+/**
+* @Synopsis  Allocates a buffer and fills it with a repeated value
+*
+* @Param buffer Address of the buffer to be allocated
+* @Param buffer_length Amount of memory to be allocated
+* @Param value The value to be filled into the buffer
+* @tparam data_type The data type to be used for the buffer
+* @tparam size_type The data type used for size calculations
+*/
+/* ----------------------------------------------------------------------------*/
+template <typename data_type,
+          typename size_type>
+void allocValueBuffer(
+        data_type ** buffer,
+        const size_type buffer_length,
+        const data_type value) {
+    cudaMalloc(buffer, buffer_length*sizeof(data_type));
+    thrust::copy(
+            thrust::device,
+			thrust::make_constant_iterator(value),
+			thrust::make_constant_iterator(value) + buffer_length,
+            *buffer);
+}
+
+/* --------------------------------------------------------------------------*/
+/**
+* @Synopsis  Allocates a buffer and fills it with a sequence
+*
+* @Param buffer Address of the buffer to be allocated
+* @Param buffer_length Amount of memory to be allocated
+* @tparam data_type The data type to be used for the buffer
+* @tparam size_type The data type used for size calculations
+*/
+/* ----------------------------------------------------------------------------*/
+template <typename data_type,
+          typename size_type>
+void allocSequenceBuffer(
```

Comment:
Could this be replaced with `thrust::sequence`? https://thrust.github.io/doc/group__transformations.html#ga08c1dd7914c155d5fed6458330af3443